fdbserver/ApplyMetadataMutation.cpp

```diff
@@ -385,7 +401,7 @@ class ApplyMetadataMutationsImpl {
 	void checkSetStorageCachePrefix(MutationRef m) {
 		if (!m.param1.startsWith(storageCachePrefix))
 			return;
-		if (cacheInfo) {
+		if (cacheInfo || forResolver) {
```

Comment:
Why it is OR logic here, not AND? If cacheInfo is set in commitProxy, will it still generate private mutation?

Reply:
At commit proxies, `cacheInfo` is never a nullptr, i.e., current behavior is to enter this block at proxies. Here, Resolver always has `cacheInfo` with nullptr value, and we still want to generate private mutations for this path.

Reply:
This implementation is based on the fact that `cacheInfo` is never nullptr. If `cacheInfo` can be set to nullptr in the future, here we will generate incorrect private mutation. I think it worth document here to reduce some confusion.

Reply:
Agreed. I added a comment here.